app/logbook/olog/ui/src/main/java/org/phoebus/logbook/olog/ui/AttachmentsPreviewController.java

```diff
@@ -120,13 +122,22 @@ public void changed(ObservableValue<? extends Attachment> observable, Attachment
                         defaultApp.create(attachment.getFile().toURI());
                         return;
                     }
+
                     // If not internal apps are found look for external apps
                     String fileName = attachment.getFile().getName();
                     String[] parts = fileName.split("\\.");
                     if(parts.length == 1 || !ApplicationService.getExtensionsHandledByExternalApp().contains(parts[parts.length - 1])){
-                        // If there is no app configured for the file type, show an error message and return.
-                        ExceptionDetailsErrorDialog.openError(Messages.PreviewOpenErrorTitle, Messages.PreviewOpenErrorBody, null);
-                        return;
+                        // If there is no app configured for the file type, then use the default configured for the OS/User
+                        try {
+                            if(Desktop.isDesktopSupported()) {
```

Comment:
Desktop is from java.awt, while we're running under JavaFX.
That can result in hangups.

The display builder falls back from the ApplicationLauncherService to the JavaFX  `Application.getHostServices()`, 
https://github.com/ControlSystemStudio/phoebus/blob/ab909e323bcfe377511f5cb5df20dda5c746a22b/app/display/representation-javafx/src/main/java/org/csstudio/display/builder/representation/javafx/JFXRepresentation.java#L1021

Reply:
Thanks @kasemir, could explain the hand I spotted on CentOS7

@cjenkscybercom, can you please update according to @kasemir 's input? I'll run a new round of tests.

Reply:
We've definitely seen hangups with calls to  AWT `Desktop`, see https://github.com/ControlSystemStudio/phoebus/issues/433

Mind you I don't think we can predict what happens inside HostServices. The documentation mentions opening the file in a web browser,  https://openjfx.io/javadoc/13/javafx.graphics/javafx/application/HostServices.html, and I think that also tends to work for PDF documents. But will it open MS Word, Excel, ...?

Reply:
The idea is not to provide an exhaustive solution, but to support a "best effort". I'm sure users do not expect MS Office files in general to open on Linux. As for the behavior on Windows/Mac, let's try it out...

Reply:
Hmmm...I'll give this a shot, thanks @kasemir !

Reply:
Do we have a standard way of getting access to the Application or HostServices?
The controller doesn't extend Application or have it/HostServices available, and Application ofc doesn't have a static getHostServices method?